lib/elixir/lib/kernel/lexical_tracker.ex

```diff
@@ -46,8 +60,8 @@ defmodule Kernel.LexicalTracker do
 
   # Starts the tracker and returns its pid.
   @doc false
-  def start_link(dest) do
-    :gen_server.start_link(__MODULE__, dest, [])
+  def start_link(dest, parent) do
```

Comment:
`Kernel.LexicalTracker.start_link` now takes a parameter `parent`, which is another tracker pid that will get all lexical data forwarded from this tracker.

Before now, all modules in a file would actually end up using the same tracker, which resulted in all modules after the first containing all of the references of the ones defined above them. Because the data was only used to determine what the dependencies for compilation were, this was not an issue.

Now, there is a tree of trackers, and compiling a file with multiple modules results in a file-scope tracker which receives all data about the modules, as well as individual module-scoped trackers. This means that the file-scope tracker can determine whether there should be warnings for imports in the file scope, and that module-scope trackers only contain calls within their module's definition.